### PR TITLE
Multiselect user filters

### LINKED ISSUE #1645: Ability to exclude in user filters
### Is your feature request related to a problem?

_No response_

### What is your feature request?

Currently, you can see stats (History, Graphs, etc.) for "All users" or for a specific one. 
Graphs
History

It would be nice if you could see stats for all users except some of them. A possible implementation would be a checkbox next to the username.
Possible implementation
 

### Are there any workarounds?

I see there are other features, kind of related to this.
[User groups](https://github.com/Tautulli/Tautulli/issues/1557)

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by zdimension**
## Description

Previously, user filtering was limited to seeing stats for either one user or all users. This was seen as [an](https://github.com/Tautulli/Tautulli/issues/1645) [issue](https://www.reddit.com/r/Tautulli/comments/g7ij28/exclude_users_from_graphs/) especially for the quite common case of one Plex user being a "power user" and using Plex a lot more than the others. Excluding an user, or, more generally, filtering for multiple users, appears as something useful.

This PR replaces the old, simple \<select>-based filter by one using Bootstrap-select that allows multiple selection.

The changes in the frontend are limited to graphs.html, the main stylesheet, and the addition of the Bootstrap-select vendor files.

The changes in the backend are limited to graphs.py (the graph APIs) and webserve.py (the exposed API functions).

### Screenshot

|Before|After|
|--|--|
|image|image|

### Issues Fixed or Closed

- Fixes #1645

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the docstring for new or existing methods

(this PR should be **squashed** when merged)

### CODE REVIEW

File: data/interfaces/default/graphs.html [hunk 1]
```diff
@@ -373,14 +373,35 @@ <h4><i class="fa fa-calendar"></i> Total <span class="yaxis-text" style="text-tr
...
                 data.forEach(function(item) {
                     select.append('<option value="' + item.user_id + '">' +
                         item.friendly_name + '</option>');
+                    by_id[item.user_id] = item.friendly_name;
                 });
+                select.selectpicker({
+                    countSelectedText: function(sel, total) {
+                        if (sel === 0 || sel === total) {
+                            return 'All users';
+                        } else if (sel > 3) {
```
Comment (JonnyWong16) [inline data/interfaces/default/graphs.html]: 2 usernames can already overflow the container.

```suggestion
                        } else if (sel > 1) {
```

File: plexpy/graphs.py [hunk 2]
```diff
@@ -165,17 +161,26 @@ def get_total_plays_per_day(self, time_range='30', y_axis='plays', user_id=None,
                   'series': series_output}
         return output
 
+    def make_user_cond(self, user_id):
```
Comment (JonnyWong16) [inline plexpy/graphs.py]: Move this method to the bottom just to organize better.

Also make the method private.
```suggestion
    def _make_user_cond(self, user_id):
```

File: plexpy/graphs.py [hunk 3]
```diff
@@ -165,17 +161,26 @@ def get_total_plays_per_day(self, time_range='30', y_axis='plays', user_id=None,
...
+        """
+        Expects user_id to be a comma-separated list of ints.
+        """
+        user_cond = ''
+        if session.get_session_user_id() and user_id and user_id != str(session.get_session_user_id()):
+            user_cond = 'AND session_history.user_id = %s ' % session.get_session_user_id()
+        elif user_id:
+            user_ids = [id.strip() for id in user_id.split(',')]
```
Comment (JonnyWong16) [inline plexpy/graphs.py]: There is a helper function for this.
```suggestion
            user_ids = helpers.split_strip(user_id)
```

**Review by JonnyWong16 — CHANGES_REQUESTED** [review]
Looks good. Just a couple minor changes.

Could you also implement this on the history page? The backend already supports a comma-separated list of user IDs, so the new dropdown just needs to be added to the UI.

**Comment by zdimension** [comment]
> Looks good. Just a couple minor changes.
> 
> Could you also implement this on the history page? The backend already supports a comma-separated list of user IDs, so the new dropdown just needs to be added to the UI.

I pushed the minor changes, and added the picker to the history page (tested locally, works perfectly)

**Review by JonnyWong16 — APPROVED** [review]

**Pull request merged**